### PR TITLE
Fhac 170

### PR DESCRIPTION
**Pull request by TabassumJafri**
FHAC-170: HPD client implementation and the associated testcases for querying HPD server

### CODE REVIEW

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/FederatedRequestData.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * and open the template in the editor.
+ */
+
+package gov.hhs.onc.hpdclient;
+
+import java.io.Serializable;
+
+/**
+ *
+ * @author Wilmertech
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/FederatedRequestData.java]: who is wilmerTech?  do we need to credit this to another source?

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/FederatedRequestData.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/FederatedRequestData.java]: License needs update.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+ *
+ * @author tjafri
+ */
+public class HPDClient {
+
+    private static ObjectFactory dsmlBasedObjectFactory = null;
+    private static ProviderInformationDirectoryService pdService = null;
+    private static ProviderInformationDirectoryPortType pdPortType = null;
+
+    private static final String DN_HC_PROFESSIONAL = "ou=HcProfessional,o=dev.provider-directories.com,dc=hpd";
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java]: I think it was better when these values were in a properties file.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        System.out.println("professional givenName Thomas");
+        System.out.println("organization businessCategory MRI");
+        System.out.println("membership hpdHasAProvider uid=2.16.840.1.113883.3.4295:provider1,ou=HcProfessional,o=dev.provider-directories.com,dc=hpd");
+        System.out.println("services hpdServiceId 6");
+        System.out.println("credential credentialNumber 1");
+        System.out.println("relationship cn JonesPracticeGroup");
+    }
+
+    private static void searchQuery(String[] arg, String dn) {
+        String requestId = UUID.randomUUID().toString();
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java]: I'd pull out everything in this method and the called private methods and put them in a service class with an interface.  Should treat the command line call as a view and separate the actual functional piece so we could overlay a gui or call it as a lib.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java]: Add connect license header.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    private static final String DEREF_ALIASES = "pdti.server.derefAliases";
+    private static final String WSDL_URL = "pdti.server.wsdlurl";
+    private static final String PROPERTY_FILENAME = "config.properties";
+
+    public HPDServiceImpl() throws IOException {
+
+        prop = new Properties();
+        InputStream inputStream = null;
+        try {
+            //inputStream = this.getClass().getClassLoader().getResourceAsStream(PROPERTY_FILENAME);
```
Comment (jsmith2) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java]: looks like dead code that is no longer necessary.  Please delete comments.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ * @author tjafri
+ */
+public class HPDServiceImpl implements HPDService {
+
+    private ObjectFactory dsmlBasedObjectFactory;
+    private ProviderInformationDirectoryService pdService;
+    private ProviderInformationDirectoryPortType pdPortType;
+    private Properties prop;
+    private static final String DN_HC_PROFESSIONAL = "pdti.server.dnHcProfessional";
```
Comment (nareshcgi) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/service/HPDServiceImpl.java]: You can remove the pdti prefix from the property name. We can potentially use the same code to test some other implementation also.

File: HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.Marshaller;
+import javax.xml.namespace.QName;
+import oasis.names.tc.dsml._2._0.core.BatchResponse;
+import oasis.names.tc.dsml._2._0.core.ErrorResponse;
+import oasis.names.tc.dsml._2._0.core.SearchResponse;
+
+/**
+ *
```
Comment (nareshcgi) [inline HPD-WS-Client/src/main/java/gov/hhs/onc/hpdclient/HPDClient.java]: Add some class level comments

**Comment by nareshcgi** [comment]
Reviewed the PR and looks good.

**Pull request merged**